plugins/generator-1.17.1/forge-1.17.1/templates/elementinits/gamerules.java.ftl

```diff
@@ -0,0 +1,84 @@
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2021, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+
+/*
+ *    MCreator note: This file will be REGENERATED on each build.
+ */
+
+package ${package}.init;
+
+public class ${JavaModName}GameRules {
+
+	<#list gamerules as gamerule>
+	<#if gamerule.type == "Number">
+	public static GameRules.Key<GameRules.IntegerValue> ${gamerule.getModElement().getRegistryNameUpper()};
+	<#else>
+	public static GameRules.Key<GameRules.BooleanValue> ${gamerule.getModElement().getRegistryNameUpper()};
+	</#if>
+	</#list>
+
+	public static void load() {
+		<#list gamerules as gamerule>
+		<#if gamerule.type == "Number">
+		${gamerule.getModElement().getRegistryNameUpper()} = GameRules.register("${gamerule.getModElement().getRegistryName()}",
+				GameRules.Category.${gamerule.category}, create(${gamerule.defaultValueNumber}));
+		<#else>
+		${gamerule.getModElement().getRegistryNameUpper()} = GameRules.register("${gamerule.getModElement().getRegistryName()}",
+				GameRules.Category.${gamerule.category}, create(${gamerule.defaultValueLogic});
+		</#if>
+		</#list>
+	}
+
+	private static GameRules.Type<GameRules.IntegerValue> create(int defaultValue) {
+		try {
+			Method createGameRuleMethod = ObfuscationReflectionHelper.findMethod(GameRules.IntegerValue.class, "create", int.class);
+			createGameRuleMethod.setAccessible(true);
+			return (GameRules.Type<GameRules.IntegerValue>) createGameRuleMethod.invoke(null, defaultValue);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+
+	private static GameRules.Type<GameRules.BooleanValue> create(boolean defaultValue) {

```

Comment:
The same comment applies to this method.

Reply:
:heavy_check_mark: Fixed

Reply:
I think a new Java method you can use here will be better. Check these links for more info. 
https://github.com/MCreator/MCreator/blob/9eb8aa8936548e66acf7b7e1521d462008c919b6/plugins/generator-1.17.1/forge-1.17.1/templates/modbase/mod.java.ftl#L40
https://github.com/MCreator/MCreator/commit/6e06b3fdf03b1dab5f16922a8fa7e51ed77d9672#diff-7f2d9a7234d68266489668f091a550f138c2b7ca2c17d01f0bc04ce4952acbd3R143